public/app/features/alerting/unified/RuleEditor.test.tsx

```diff
@@ -0,0 +1,78 @@
+import { Route } from 'react-router';
+import { selectOptionInTest } from 'test/helpers/selectOptionInTest';
+import { render, screen, userEvent } from 'test/test-utils';
+
+import { setAppEvents } from '@grafana/runtime';
+import appEvents from 'app/core/app_events';
+import RuleEditor from 'app/features/alerting/unified/RuleEditor';
+import { setupMswServer } from 'app/features/alerting/unified/mockApi';
+import { grantUserPermissions, mockDataSource } from 'app/features/alerting/unified/mocks';
+import { setupDataSources } from 'app/features/alerting/unified/testSetup/datasources';
+import { DataSourceType } from 'app/features/alerting/unified/utils/datasource';
+import { AccessControlAction } from 'app/types';
+
+setupMswServer();
+
+// Required to make sure that loading the datasource plugin does not fail
+// see public/app/plugins/datasource/loki/module.ts
+setAppEvents(appEvents);
+
+const dataSources = {
+  loki: mockDataSource(
+    {
+      type: DataSourceType.Loki,
+      name: 'loki',
+      uid: 'oKflPJ2GF38UCq',
+      id: 1,
+      isDefault: true,
+      jsonData: {
+        manageAlerts: true,
+      },
+    },
+    {
+      alerting: true,
+      // needed to make the correct plugin components load
+      module: 'core:plugin/loki',
```

Comment:
This is very specific - it maps to built in plugins in https://github.com/grafana/grafana/blob/e93cbc361a39de0fa84a946652f7162e8c7472dc/public/app/features/plugins/built_in_plugins.ts#L80